spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreResultsIterable.java

```diff
@@ -50,4 +61,20 @@ public Cursor getCursor() {
 	public Iterable<T> getIterable() {
 		return this.iterable;
 	}
+
+	public Optional<Boolean> getHasNextPage() {
```

Comment:
why not just
`public boolean hasNextPage()`?

Reply:
@meltsufin Because it is not always there. We can only compute it for `StructuredQuery` queries, and we can't compute it for GQL queries. So we need to represent 3 values - true, false, empty.

Reply:
@dmitry-s I must be missing something here. Both, `GqlQuery` and `StructuredQuery`, support cursors. So, I would image that it's possible to try to retrieve the next item using the cursor to determine `hasNext()`. How are they different in this respect?
Also, if there is a fundamental difference, then we should probably have two different classes for their results. So, maybe extend `DatastoreResultsIterable` and add `hasNextPage()`?

Reply:
The problem with `GqlQuery` is we need to set limit to 1 to avoid fetching the whole next page, and `GqlQuery` doesn't allow to change that.

Reply:
So, for `GqlQuery` we would need to fetch the whole page? Is there a problem with that other than the potential performance overhead?